sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,264 @@
+| `CloudBlobClient` | `BlobClient` |
+| `CloudBlockBlob` | `BlockBlobClient` |
+
+### Creating a Container
+
+v11
+```csharp
+// Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+CloudBlobContainer cloudBlobContainer =
+    cloudBlobClient.GetContainerReference("yourcontainer");
+await cloudBlobContainer.CreateAsync();
+```
+
+v12
+```csharp
+// Create a BlobServiceClient object which will be used to create a container client
+BlobServiceClient blobServiceClient = new BlobServiceClient(connectionString);
+// Create a unique name for the container
+string containerName = "yourcontainer";
+// Create the container and return a container client object
+BlobContainerClient containerClient = await blobServiceClient.CreateBlobContainerAsync(containerName);
+```
+
+Summary: In v11, a `CloudBlobClient` was required to get a reference to the desired blob container. In v12, the intermediate step `cloudBlobClient.GetContainerReference("yourcontainer")` was removed.
+
+
+### Uploading Blobs to a Container
+
+v11
+```csharp
+// Assumes cloudBlobContainer already contains a reference to the container.
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+// Get a reference to the blob address, then upload the file to the blob.
+CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(filename);
+await cloudBlockBlob.UploadFromFileAsync(localFilePath);
+```
+
+v12
+```csharp
+// Assumes containerClient already contains a reference to the container.
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+// Get a reference to a blob
+BlobClient blobClient = containerClient.GetBlobClient(filename);
+// Open the file and upload its data
+using FileStream uploadFileStream = File.OpenRead(localFilePath);
+await blobClient.UploadAsync(uploadFileStream, overwrite: true);
+uploadFileStream.Close();
+```
+
+Summary: In v11, a file path was used to upload a blob. In v12, `Stream` is used to upload a blob's content.
+
+
+### Downloading Blobs from a Container
+
+v11
+```csharp
+// Assumes you have already created a reference to the blob via blobClient
+// downloadFilePath should be the path to the intended file to download the blob to
+await cloudBlockBlob.DownloadToFileAsync(downloadFilePath, FileMode.Create);
+```
+
+v12
+```csharp
+// Assumes you have already created a reference to the blob via blobClient
+// downloadFilePath should be the path to the intended file to download the blob to
+BlobDownloadInfo download = await blobClient.DownloadAsync();
+using (FileStream downloadFileStream = File.OpenWrite(downloadFilePath))
+{
+    await download.Content.CopyToAsync(downloadFileStream);
+    downloadFileStream.Close();
+}
+```
+
+### Listing Blobs in a Container
```

Comment:
We didn't force people to manually iterate over pages in v11. We just had separate methods for a lazy enumerable vs getting each page on your own. I think it's actually important to highlight both mechanisms for this section, because manually handling individual pages is a bit complex in v12, and we'll want to give a direct migration for that as well.